frontend/src/pages/TermPlanner/HideYearTooltip.jsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { Tooltip } from "antd";
+import { notification } from "antd";
+import { IoIosEyeOff } from "react-icons/io";
+import { useSelector, useDispatch } from "react-redux";
+import { plannerActions } from "../../actions/plannerActions";
+
+function HideYearTooltip({ year }) {
+  const { hidden, numYears } = useSelector((state) => state.planner);
+  const dispatch = useDispatch();
+
+  const showCannotHideAllYearsNotification = () => {
+    notification.open({
+      type: "error",
+      message: "Something's not right",
+      description: "You cannot hide all years in your term planner",
+      duration: 2,
+    });
+  };
+
+  const hideYear = (year) => {
+    let i = 0;
+    for (const [key, value] of Object.entries(hidden)) if (value) i++;
```

Comment:
there is nothing wrong with this line, it just facinates me lol

Reply:
I've now refined this a bit using Object.values()